spec/models/section_spec.rb

```diff
@@ -104,9 +106,18 @@ def key_classes_for(hash)
     )
   }
 
-  describe '.find' do
-    let(:manual) { double(:manual, slug: 'manual-slug') }
+  before do
+    allow(SlugGenerator).to receive(:new).with(prefix: manual_slug).and_return(slug_generator)
+  end
+
+  describe '#initialize' do
+    it 'instantiates a slug generator using the manual slug' do
+      expect(SlugGenerator).to receive(:new).with(prefix: manual_slug)
+      subject
+    end
```

Comment:
Minor: I'm not super keen on this example - specifically **expecting** the call to `SlugGenerator.new`. I think it's probably enough that we're stubbing the method in order to inject a mock object and that other examples depend on that injection, i.e. they would fail if a `SlugGenerator` was not instantiated. Having said that, maybe no other examples do depend on it the injection of the mock.